File: clojure/bob/bob.clj
(ns bob 
  (:require [clojure.core :refer :all]))

(defn- shout? [phrase]
  (and
    (= (clojure.string/upper-case phrase) phrase)
    (not= (clojure.string/lower-case phrase) phrase)))

(defn- question? [phrase]
  (= (last phrase) \?))

(defn- silence? [phrase]
  (empty? (clojure.string/trim phrase)))

(defn response-for [phrase]
  (cond
    (shout? phrase) "Whoa, chill out!"
    (question? phrase) "Sure."
    (silence? phrase) "Fine. Be that way!"
    :else "Whatever."))

File: python/bob/bob.py
def shout(sentence):
    return sentence.isupper()

def question(sentence):
    return sentence.endswith("?")

def blank(sentence):
    return sentence == ""

def hey(what):
    sentence = what.strip()
    if shout(sentence):
        return "Whoa, chill out!"
    elif question(sentence):
        return "Sure."
    elif blank(sentence):
        return "Fine. Be that way!"
    else:
        return "Whatever."
